backend/main-service/src/main/java/kkakka/mainservice/cart/application/CartService.java

```diff
@@ -29,32 +33,60 @@ public CartService(CartRepository cartRepository, CartItemRepository cartItemRep
         this.memberRepository = memberRepository;
     }
 
-    public String saveCartItem(CartRequestDto cartRequestDto) {
+    @Transactional
+    public Long saveOrUpdateCartItem(CartRequestDto cartRequestDto) {
 
         /* 테스트 데이터 */
         Optional<Member> member = memberRepository.findById(cartRequestDto.getMemberId());
 
-        /* 로그인 상태 체크 */
+        Cart cart = findOrCreateCart(member.get());
+        Optional<CartItem> cartItem = Optional.ofNullable(
+                cartItemRepository.findByMemberIdandProductId(member.get().getId(), cartRequestDto.getProductId())
+        );
 
-        /* 장바구니 존재 유무 체크 ex) 장바구니에 담아둔 아이템 있는지 */
-        Cart cart = new Cart();
-        cart = cartRepository.findByMemberId(cartRequestDto.getMemberId());
-        if (cart == null) {
-            cartRepository.save(new Cart(member.get()));
-            cart = cartRepository.findByMemberId(cartRequestDto.getMemberId());
+        // 동일 상품 이미 있으면 수량 업데이트
+        if (cartItem.isPresent()) {
+            Integer preQuantiry = cartItem.get().getQuantity();
+            cartItem.get().setQuantity(preQuantiry + cartRequestDto.getQuantity());
+            cartItemRepository.save(cartItem.get());
```

Comment:
```suggestion
            CartItem originCartItem = cartItem.get();
            Integer preQuantiry = originCartItem.getQuantity();
            originCartItem.setQuantity(preQuantiry + cartRequestDto.getQuantity());
            cartItemRepository.save(originCartItem);
```
여러 번 사용하는 변수라 따로 선언해도 좋을 것 같아요!

그리고, 장바구니 수량이 바뀌면 더해야할까요 아니면 변경을 해야할까요.. 
이건 정책적인 부분이라 생각을 해봐야겠지만, 장바구니 최초 추가 + 변경에 같은 API를 사용하고 있으니 변경이 더 자연스러운 로직이 아닐까 생각이 되네요!!

Reply:
저도 생각해보니 그냥 새로운 수량으로 변경을 하는게 로직이 훨씬 자연스러울거 같아요 
변수도 처리하겠습니다 !